scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/instances/CoderInstances.scala

```diff
@@ -0,0 +1,27 @@
+package com.spotify.scio.extra.sparkey.instances
+
+/*
+ * Copyright 2023 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.extra.sparkey.SparkeyUri
+import org.apache.beam.sdk.{coders => bcoders}
+
+trait CoderInstances {
+  implicit def sparkeyUriCoder[T <: SparkeyUri]: Coder[T] =
+    Coder.beam(bcoders.SerializableCoder.of(classOf[SparkeyUri])).asInstanceOf[Coder[T]]
```

Comment:
If `ShardedSparkeyUri` and `SparkeyUri` were in the same file, we could just make them `sealed traits` and let Scio derive coders automatically 👀

Reply:
It should be ok; added more context to PR description.

Reply:
I wonder what concrete type of Beam coder is resolved here?

Reply:
`SparkeyUri` but we want a `T`